### DISCUSSION TITLE
Access the config

### DISCUSSION THREAD

**Discussion by saberone** [dotnet/yarp#2559] [General]
The application that hosts Yarp is also doing api calls to one of the backend services. I'm using a dedicated HttpClient to do that. What I would like to do is use a destination url that is configured for a named destination.

So my question. How can I access the configured destination from code?

As soon as I asked this question, my brain apparently started to work again ;)  I'm using the code below to access the configured destination.

```cs
var  proxyConfig = sp.GetService<IProxyConfigProvider>();

var conf = proxyConfig.GetConfig();
var backendAddress = conf.Clusters
                            .Where(cluster => cluster.ClusterId == "vNextApi")
                            .FirstOrDefault()
                            .Destinations["destination1"].ToString();
```

**Comment by benjaminpetit** [comment]
Glad you found the answer :)
